src/tests/functional/test_competitions.py

```diff
@@ -12,6 +13,15 @@ def setUp(self):
         self.user = UserFactory(password='test')
         self.login(self.user.username, 'test')
 
+    def current_server_time_exists(self):
+        # Get server time element
+        element = self.find('#server_time')
+        text = element.get_attribute('innerText')
+
+        # Check that the text is a valid datetime by loading it with strptime.
+        # This will raise a ValueError if the format is incorrect.
+        assert datetime.strptime(text, '%B %d, %Y, %I:%M %p %Z')
```

Comment:
The problem with this test is that if it is in a fail state, we don't get the nice pytest interpreted assertion error, we just get a value error, as the comment notes. we want to avoid that, since this type of a failure is slower to locate and fix. my preferred solution would be something like the following pseudo-code:
```python
try:
    # assert that the time displayed _is_ server time. possible problem here: I believe pretty date 
    # shifts timezones to the user's local timezone, maybe convert to unix timestamps to compare.
    assert strptime(...) == timezone.now()
except ValueError:
    assert False, "Date not displayed in the correct format"
```
This tests that the server time is present, correct, _and_ in the proper format.

Reply:
I'll check if we can gain access to a js variable in selenium, so we can make failure states of this test more informative. It will go something like the following:

1. Check for `CURRENT_SERVER_TIME` variable in js directly.
2. Check that there is content in the `#server_time` element (`len(text) > 0`).
3. Check that `text` is formatted properly as a datestring by loading it with `strptime`.